### ISSUE TITLE
Add Style/StringConcatenation support for << with string literals?

### ISSUE THREAD

**Issue by vlad-pisanov** [rubocop/rubocop#10918]
`Style/StringConcatenation` cop detects string concatenation using `+` like so:

```ruby
'hello ' + name
# corrects to interpolation:
"hello #{name}"
```

It seems this cop could also detect concatenation using `<<` in a similar way if the left-hand operand is a literal:
```ruby
'hello ' << name
# would also correct to:
"hello #{name}"
```

**Comment by ydakuka** [comment]
I think it should be closed due to https://github.com/rubocop/rubocop/pull/11101#issuecomment-1293218980
